src/Simplex/Messaging/Transport.hs

```diff
@@ -325,30 +348,45 @@ data ClientHandshake = ClientHandshake
 
 instance Encoding ClientHandshake where
   smpEncode ClientHandshake {smpVersion, keyHash, authPubKey} =
-    smpEncode (smpVersion, keyHash) <> encodeAuthPubKey smpVersion authPubKey
+    smpEncode (smpVersion, keyHash) <> encodeAuthEncryptCmds smpVersion authPubKey
   smpP = do
     (smpVersion, keyHash) <- smpP
     -- TODO drop SMP v6: remove special parser and make key non-optional
-    authPubKey <- authPubKeyP smpVersion
+    authPubKey <- authEncryptCmdsP smpVersion smpP
     pure ClientHandshake {smpVersion, keyHash, authPubKey}
 
 instance Encoding ServerHandshake where
-  smpEncode ServerHandshake {smpVersionRange, sessionId, authPubKey} =
-    smpEncode (smpVersionRange, sessionId) <> encodeAuthPubKey (maxVersion smpVersionRange) authPubKey
+  smpEncode ServerHandshake {smpVersionRange, sessionId, authPubKey, certChain} =
+    B.concat
+      [ smpEncode (smpVersionRange, sessionId),
+        encodeAuthEncryptCmds maxV $ encodeChain <$> certChain,
+        encodeAuthEncryptCmds maxV $ C.SignedObject <$> authPubKey
+      ]
```

Comment:
you cannot make it independently of the version, it won't work with the old clients who will only be able to parse smpVersionRange and sessionId

Reply:
ah, this checks version twice... this seems unnecessarily confusing, needs to change somehow

Reply:
I thought maybe we will add chain and pubkey in different versions. Better put them under one check ofc, if possible.